File: js/programmers/level.1/prgm_140108_1.js
function solution(s) {
  let firstChar = "";
  let sameCount = 0;
  let deffCount = 0;
  let answer = 0;

  for (let char of s) {
    if (!firstChar) firstChar = char;

    if (firstChar === char) sameCount++;
    else deffCount++;

    if (sameCount === deffCount) {
      answer++;
      sameCount = 0;
      deffCount = 0;
      firstChar = "";
    }
  }
  if (firstChar) answer++;
  return answer;
}

File: python/programmers/level.1/prgm_140108_1.py
def solution(s):
    x = ""
    x_count = 0
    y_count = 0
    answer = 0

    for char in s:
        if not x: x = char
        if x == char: x_count += 1
        else: y_count += 1

        if x_count == y_count:
            answer += 1
            x_count, y_count, x = 0, 0, ""

    if x: answer += 1
    
    return answer